docs/data/material/getting-started/versions/ReleasedVersions.js

```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import Table from '@mui/material/Table';
+import TableBody from '@mui/material/TableBody';
+import TableCell from '@mui/material/TableCell';
+import TableRow from '@mui/material/TableRow';
+import Typography from '@mui/material/Typography';
+import { Link } from '@mui/docs/Link';
+
+const GITHUB_RELEASE_BASE_URL = 'https://github.com/mui/material-ui/releases/tag/';
+
+const VersionsContext = React.createContext(null);
+
+if (process.env.NODE_ENV !== 'production') {
+  VersionsContext.displayName = 'VersionsContext';
+}
+
+export default function ReleasedVersions() {
+  const versions = React.useContext(VersionsContext);
+
+  return (
+    <Table sx={{ minHeight: 33 * 11, overflow: 'auto', width: '100%' }}>
+      <TableBody>
+        {versions.map((doc) => (
```

Comment:
This is not working at the moment — it returns null. I don't really know why, as this is exactly the same code available in the other (duplicated) component, which you can find in `src/pages/versions/ReleasedVersion`. I just put the useContext-related code in this same file.

Reply:
I added a `versions &&` check in my last commit just to get the deploy preview link working. The comment above is still valid—not sure why it is returning null 😬

Reply:
See this comment: https://github.com/mui/material-ui/pull/42438#discussion_r1618491743

We need to wrap the `docs/pages/material-ui/getting-started/versions.js` page with `VersionsContext.Provider` and get the value for the provider from  `Page.getInitialProps`. Same as we're doing in `/docs/pages/versions.js`.

```js
export default function Page(props) {
  const { versions } = props;
  return (
    <VersionsContext.Provider value={versions}>
      ...
    </VersionsContext.Provider>
  );
}

Page.getInitialProps = async () => {
  ...

  return { versions: ... };
}

Reply:
Uhm 🤔 there's none of this code on the existing `ReleasedVersions` component, though — aren't we already feeding the `versions` prop via the `VersionsContext`? I'm confused 😅

Reply:
`ReleasedVersions` reads the `VersionsContext` that is wrapping the page. In the original `versions.js` page, you see the code I pasted in my previous comment. `<VersionsContext.Provider>` defines the context value that `ReleasedVersions` later reads.

In this PR, there's no `<VersionsContext.Provider>` wrapping the contents of the new `versions.js` page, that's why `ReleasedVersions` can't read the context.

Reply:
Oh, I've totally overseen this file: https://github.com/mui/material-ui/blob/next/docs/pages/versions.js 🤦 Any recommendations for dealing with this? Like, should we move all this code closer to the newer Versions page?

Reply:
Alright — that's done! 👍